homeassistant/components/ccl/__init__.py

```diff
@@ -0,0 +1,107 @@
+"""The CCL Electronics integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from aioccl import CCLDevice, CCLServer
+from aioccl.exception import CCLDeviceRegistrationException
+from aiohttp import web
+from aiohttp.hdrs import METH_POST
+
+from homeassistant.components import webhook
+from homeassistant.const import CONF_WEBHOOK_ID, EVENT_HOMEASSISTANT_STOP, Platform
+from homeassistant.core import HomeAssistant, callback
+
+from .const import DOMAIN, NAME
+from .coordinator import CCLConfigEntry, CCLCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: CCLConfigEntry) -> bool:
+    """Set up a config entry for a single CCL device."""
+    if hasattr(entry, "runtime_data"):
+        assert isinstance(entry.runtime_data, CCLCoordinator)
+        _LOGGER.debug(
+            "Retrying connecting to device with webhook ID %s",
+            entry.data[CONF_WEBHOOK_ID],
+        )
+        device = entry.runtime_data.device
+        coordinator = entry.runtime_data
+
+    else:
+        device = CCLDevice(entry.data[CONF_WEBHOOK_ID])
+        coordinator = entry.runtime_data = CCLCoordinator(hass, device, entry)
+
+        async def register_webhook() -> None:
+            """Register webhook for the device."""
+
+            def handle_webhook(
+                hass: HomeAssistant, webhook_id: str, request: web.Request
+            ) -> Any:
+                """Handle incoming requests from CCL devices."""
+                return CCLServer.handler(request)
+
+            webhook_url = webhook.async_generate_url(
+                hass,
+                entry.data[CONF_WEBHOOK_ID],
+                allow_external=False,
+                allow_ip=True,
+            )
+
+            webhook_name = "CCL Electronics"
+            if entry.title != NAME:
+                webhook_name = f"{NAME} {entry.title}"
+
+            webhook.async_register(
+                hass,
+                DOMAIN,
+                webhook_name,
```

Comment:
```suggestion
            webhook.async_register(
                hass,
                DOMAIN,
                f"{NAME} {entry.title}",
```
let's try to keep in concise